src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +973,73 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
+                self._insufficient_capacity_compute_resources[compute_resource] = {
+                    "ice_time": self._current_time,
+                    "error_code": node.error_code,
+                }
+
+    def reset_timeout_expire_compute_resources(self, ice_nodes):
+        """Reset compute resources which insufficient_capacity_timeout expire."""
+        # Find insufficient_capacity_timeout compute resources
+        if not self._insufficient_capacity_compute_resources:
+            return
+        log.info(
+            "Insufficient capacity compute resources: %s, compute resources will be reset after "
+            "insufficient capacity timeout(%s seconds) expire",
+            self._insufficient_capacity_compute_resources,
+            self._config.insufficient_capacity_timeout,
+        )
+        timeout_expire_cr = {}
+        for cr, attribute in self._insufficient_capacity_compute_resources.copy().items():
+            if time_is_up(attribute.get("ice_time"), self._current_time, self._config.insufficient_capacity_timeout):
+                timeout_expire_cr[cr] = self._insufficient_capacity_compute_resources.pop(cr)
+
+        # Reset nodes which insufficient capacity timeout expire
+        if timeout_expire_cr:
+            logging.info(
+                f"Reset the following compute resources because insufficient capacity timeout expire: {timeout_expire_cr}"
+            )
+            nodes_to_power_down = [node for node in ice_nodes if node.get_compute_resource_name() in timeout_expire_cr]
+
+            if nodes_to_power_down:
+                log.info(
+                    "Reset the following ICE nodes because insufficient capacity timeout expire for the compute resource: %s",
+                    print_with_count(nodes_to_power_down),
+                )
+                set_nodes_power_down(
+                    [node.name for node in nodes_to_power_down],
+                    reason="Reset ice nodes after insufficient_capacity_timeout expire.",
+                )
+
+    def _set_ice_crs_to_down(self, active_nodes):
+        if not self._insufficient_capacity_compute_resources:
+            return
+        nodes_to_down = {}
+        for node in active_nodes:
+            error_code = self._insufficient_capacity_compute_resources.get(node.get_compute_resource_name(), {}).get(
+                "error_code"
+            )
+            if error_code and node.is_powered_down_and_not_ice_cr():
+                nodes_to_down[error_code] = nodes_to_down.get(error_code, []) + [node.name]
+        if nodes_to_down:
+            for error_code, node_list in nodes_to_down.items():
+                log.info(
+                    "Setting following nodes belongs to insufficient capacity computeresources into DOWN state: %s",
+                    print_with_count(node_list),
+                )
+                set_nodes_down(node_list, reason=f"(Code:{error_code})Node belongs to ICE compute resource")
```

Comment:
Node belongs to ICE compute resource -> Temporarily disabling node due to insufficient capacity

Reply:
Done